test/lit/ctor-eval/gc-cycle-multi.wast

```diff
@@ -0,0 +1,145 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+;; RUN: foreach %s %t wasm-ctor-eval --ctors=test1,test2,test3 --kept-exports=test1,test2,test3 --quiet -all -S -o - | filecheck %s
+
+;; Similar to gc-cycle.wast, but now there are three exported ctors that we
+;; call.
+
+(module
+ ;; Multiple independent cycles.
+ ;; CHECK:      (type $A (struct (field (mut (ref null $A))) (field i32)))
+ (type $A (struct (field (mut (ref null $A))) (field i32)))
+
+ ;; CHECK:      (type $none_=>_none (func))
+
+ ;; CHECK:      (type $none_=>_i32 (func (result i32)))
+
+ ;; CHECK:      (global $ctor-eval$global_8 (ref $A) (struct.new $A
+ ;; CHECK-NEXT:  (ref.null none)
+ ;; CHECK-NEXT:  (i32.const 30)
+ ;; CHECK-NEXT: ))
+
+ ;; CHECK:      (global $c (mut (ref null $A)) (global.get $ctor-eval$global_8))
+
+ ;; CHECK:      (global $ctor-eval$global_7 (ref $A) (struct.new $A
+ ;; CHECK-NEXT:  (ref.null none)
+ ;; CHECK-NEXT:  (i32.const 10)
+ ;; CHECK-NEXT: ))
+
+ ;; CHECK:      (global $a (mut (ref null $A)) (global.get $ctor-eval$global_7))
+ (global $a (mut (ref null $A)) (ref.null $A))
+ ;; CHECK:      (global $ctor-eval$global_6 (ref $A) (struct.new $A
+ ;; CHECK-NEXT:  (ref.null none)
+ ;; CHECK-NEXT:  (i32.const 20)
+ ;; CHECK-NEXT: ))
+
+ ;; CHECK:      (global $b (mut (ref null $A)) (global.get $ctor-eval$global_6))
+ (global $b (mut (ref null $A)) (ref.null $A))
+ (global $c (mut (ref null $A)) (ref.null $A))
+
+ (func $makeCycle (param $i i32) (result (ref $A))
+  (local $x (ref $A))
+  (local.set $x
+   (struct.new $A
+    (ref.null $A)
+    (local.get $i)
+   )
+  )
+  (struct.set $A 0
+   (local.get $x)
+   (local.get $x)
+  )
+  (local.get $x)
+ )
+
+ (func "test1"
+  (global.set $a
+   (call $makeCycle
+    (i32.const 10)
+   )
+  )
+ )
+
+ (func "test2"
+  (global.set $b
+   (call $makeCycle
+    (i32.const 20)
+   )
+  )
+ )
+
+ (func "test3"
+  (global.set $c
+   (call $makeCycle
+    (i32.const 30)
+   )
+  )
+ )
+
+ (func "keepalive" (result i32)
```

Comment:
If you give these functions `$name`s, then the test update script would be able to match them up with the output better (assuming wasm-ctor-eval does not change the names).

Reply:
Good idea, I did that to the `keepalive` functions now (the others are actually replaced as we process them).